packages/environment-check/src/logger.ts

```diff
@@ -0,0 +1,30 @@
+import type { Logger, ResultMessage } from './types';
```

Comment:
Could you please reuse the logger module `@sap-ux/logger`?

Reply:
> Could you please reuse the logger module `@sap-ux/logger`?

The logger in this case does not directly output anything, but collects the messages while executing several checks and [outputs it later](https://github.com/SAP/open-ux-tools/pull/628/files#diff-aacc7ccf720328c0fc8469c6a256d217767054b41b040766cabea867e73a777bR70) or [parses it into markdown](https://github.com/SAP/open-ux-tools/pull/628/files#diff-3e199b0492bd60a1f118415d232497ac96b74f08533b0d1f32bcbf7276f60bd6R237). 

So it is more like a message collector. To make it clear, we could rename it and use `@sap-ux/logger` in places when we really output something.

Reply:
@cianmSAP, perhaps even better would be to pass a custom transport to `@sap-ux/logger` that does the message collection.

Reply:
I have added the respective transport for envcheck to `@sap-ux/logger`: https://github.com/SAP/open-ux-tools/pull/628/commits/bd462e9d60dae8466e08164816a875e7f768bff3 @ullasholla, perhaps you can also have a look at the commit.

@cianmSAP, in envcheck I did the changes to use the logger with the newly added transport: https://github.com/SAP/open-ux-tools/pull/628/commits/9dacfd6d6d28f54a5d4e7d1a7aad48d100875657
I've adjusted the `Severity` to be aligned with `@sap-ux/logger`, here is the map 
| before | after |
| ----- | ----- |
| `error` | `error`|
| ` warn` | `warn` |
| ` log` | `info` |
| ` info` | `debug` |

Reply:
Thanks, I leave it to @ullasholla to review the changes of the `logger` module.

Reply:
Seeing this error when testing on SBAS. I'll try fix

<img width="517" alt="Screenshot 2022-08-25 at 11 29 07" src="https://user-images.githubusercontent.com/99677697/186641845-4d85ab73-66ea-4222-b579-624cd93d2e85.png">

Reply:
@cianmSAP, can't reproduce this, did you build logger?
```
cd packages/logger
pnpm build
```

Reply:
After a fresh clone and build of the repos, it works.

Reply:
@Klaus-Keller:
> I have added the respective transport for envcheck to @sap-ux/logger: https://github.com/SAP/open-ux-tools/commit/bd462e9d60dae8466e08164816a875e7f768bff3 @ullasholla, perhaps you can also have a look at the commit.

Thanks Klaus. The changes look good, the tests are correctly updated. I like the use of mixins. What's going in is essentially an `ArrayTransport` in object mode. Does it make sense to change the name? It's also useful in the store module, for example. Also, can something like this be reused? https://github.com/pustovitdmytro/winston-array-transport

Reply:
@ullasholla 
> Thanks Klaus. The changes look good, the tests are correctly updated. I like the use of mixins. What's going in is essentially an `ArrayTransport` in object mode. Does it make sense to change the name? It's also useful in the store module, for example. Also, can something like this be reused? https://github.com/pustovitdmytro/winston-array-transport

Thanks for checking, I had a look at `winston-array-transport`. It's basically the same: 
https://github.com/pustovitDmytro/winston-array-transport/blob/7deb8fd859ce1764f600bd05a1deac34f57a680f/src/index.js#L21-L29
```
    log(info, callback) {
        setImmediate(() => {
            this.emit('logged', info);
        });
        const message = info[MESSAGE];

        this.array.push(this.parser(message));
        if (this.limit && this.array.length > this.limit) this.array.shift();
        callback();
    }
```

compare to what we do:
```
    log(info: { level: string; message: string }, next: () => void) {
        this.messageCollector.push(info);
        next();
    }
```

But then, I don't think it is worth having an additional dependency which might have other dependencies and keep them up-to-date just for a couple of lines of code. I can certainly rename the `EnvcheckTransport` to `ArrayTransport`. Currently, I don't allow to pass an own reference to an array, because I thought it is better to use the logger and access the transport to read all messages than keeping an array with messages somewhere else, but I can also change that. Anyway, as one of the main contributor to `@sap-ux/logger` you can decide what you like: own transport (renamed) or external transport like `winston-array-transport`, just let me know and I do the changes.

Reply:
> But then, I don't think it is worth having an additional dependency which might have other dependencies and keep them up-to-date just for a couple of lines of code.

I agree. Right now `winston-array-transport` only depends on `winston-transport` but that could change in the future. We can keep the option of using it (or something similar) in the future, if it becomes a pain to maintain our own transport implementation.

> I can certainly rename the EnvcheckTransport to ArrayTransport

Yes, please.

> I don't allow to pass an own reference to an array, because I thought it is better to use the logger and access the transport to read all messages than keeping an array with messages somewhere else, but I can also change that.

I'd prefer passing in the array so that the code that is responsible for creating the transport also manages the array. My concern is that adding code to get the logs/messages out of the `ArrayTransport` makes the interface of the transport slightly bigger (it's a readonly property right now). However, it's not a big deal right now. I'll leave it to you. If you do keep the readonly property, could you please rename it to `logs` or something? `messageCollector` doesn't seem to fit too well. Thanks!

Reply:
@ullasholla, I generalized the envcheck transport to array transport. It also allows to pass an array into now: https://github.com/SAP/open-ux-tools/pull/628/commits/eb4dc81dd362867d0abb527e643fb417c90565ee

Reply:
Looks good, thanks